File: stack.py
# N, S = map(int, input().split())  # N: Number of operations, S: Max size of stack
# stack = []

# for _ in range(N):
#     operation = input().split()
#     if operation[0] == "push":
#         x = int(operation[1])
#         if len(stack) < S:
#             stack.append(x)
#         else:
#             print("stack overflow")
#     elif operation[0] == "pop":
#         if stack:
#             print(stack.pop())
#         else:
#             print("stack underflow")
#     elif operation[0] == "top":
#         if stack:
#             print(stack[-1])
#         else:
#             print("stack underflow")

m,n = map(int, input().split())
stack = []

for _ in range(m):
    command = input().split()
    if command[0]=="push":
        x = int(command[1])
        if len(stack) < n:
            stack.append(x)
        else:
            print("Stack overflow")
    elif command[0] == "pop":
        if stack :
            print(stack.pop())
        else :
            print("stack underflow")
    elif command[0] == 'top':
        if stack:
            print(stack[-1])
        else:
            print("stack underflow")

File: Nodejs/stackQueue/stack.js
const readline = require("readline").createInterface({
  input: process.stdin,
  output: process.stdout,
});

let list = [];

readline.on("line", (input) => {
  list.push(input);
  if (list.length === parseInt(list[0]) + 1) {
    // Ensure all input lines are read
    let [N, S] = list[0].split(" ").map(Number);
    let stack = [];

    for (let i = 1; i <= N; i++) {
      let command = list[i].split(" ");

      if (command[0] === "push") {
        let x = parseInt(command[1]);
        if (stack.length < S) {
          stack.push(x);
        } else {
          console.log("stack overflow");
        }
      } else if (command[0] === "pop") {
        if (stack.length > 0) {
          console.log(stack.pop());
        } else {
          console.log("stack underflow");
        }   
      } else if (command[0] === "top") {
        if (stack.length > 0) {
          console.log(stack[stack.length - 1]);
        } else {
          console.log("stack underflow");
        }
      }
    }

    readline.close();
  }
});
